AI/VCAI/VCAI.cpp

```diff
@@ -1148,18 +1144,40 @@ void VCAI::pickBestCreatures(const CArmedInstance * army, const CArmedInstance *
 		{
 			for(int j = 0; j < GameConstants::ARMY_SIZE; j++)
 			{
-				if(armyPtr->getCreature(SlotID(j)) == bestArmy[i] && (i != j || armyPtr != army)) //it's a searched creature not in dst SLOT
+				if(armyPtr->getCreature(SlotID(j)) == bestArmy[i] && (i != j || armyPtr != destinationArmy)) //it's a searched creature not in dst SLOT
 				{
-					if(!(armyPtr->needsLastStack() && armyPtr->stacksCount() == 1)) //can't take away last creature
-						cb->mergeOrSwapStacks(armyPtr, army, SlotID(j), SlotID(i));
+					if(!(armyPtr->needsLastStack() && armyPtr->stacksCount() == 1)) //can't take away last creature without split
+						cb->mergeOrSwapStacks(armyPtr, destinationArmy, SlotID(j), SlotID(i));
+					else
+					{	
+						//TODO: Improve logic by splitting weakest creature, instead of creature that becomes last stack
+						SlotID sourceSlot = SlotID(j);
+						int lastStackCount = armyPtr->getStackCount(sourceSlot);

```

Comment:
This should be `auto` and `getStackCount` should return `TQuantity` instead of `int`